### PR TITLE
docs: versioning strategy note

### PR DESCRIPTION
**Pull request by AuHau**
Note about Bee's versioning strategy. Feel free to comment if you want different wording or something is missing.

---
This change is [Reviewable](https://reviewable.io/reviews/ethersphere/bee/2678)

### CODE REVIEW

**Review by mrekucci — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -21,6 +21,15 @@ inability to use, or your interaction with other nodes or the software.
 
 Our documentation is hosted at https://docs.ethswarm.org.
 
+## Versioning
+
+There are two versioning schemes used in Bee that you should be aware about. The main Bee version does **NOT** follow
```
Comment (acud) [inline README.md:26]: `about` => `of`

File: README.md [hunk 2]
```diff
@@ -21,6 +21,15 @@ inability to use, or your interaction with other nodes or the software.
...
+strictly Semantic Versioning. As Bee includes number of protocols it would require bumping major version for every protocol updates.
```
Comment (acud) [inline README.md:27]: strictly -> strict
number -> a number
```
Bee hosts different peer-to-peer wire protocol implementations and individual protocol breaking changes would necessitate a bump in the major part of the version. Breaking changes are expected with bumps of the minor version component, and new, backward compatible feature as well as bug fixes with a bump of the patch component. Major version bumps are reserved for significant changes in Swarm's incentive structure.
```

File: README.md [hunk 3]
```diff
@@ -21,6 +21,15 @@ inability to use, or your interaction with other nodes or the software.
 
 Our documentation is hosted at https://docs.ethswarm.org.
 
+## Versioning
+
+There are two versioning schemes used in Bee that you should be aware about. The main Bee version does **NOT** follow
+strictly Semantic Versioning. As Bee includes number of protocols it would require bumping major version for every protocol updates. 
+In this light minor versions can include breaking changes, but patch versions will be kept only for fixes.
+
+Second set of versions that is important are the Bee's OpenAPI versions that define versioning of the Bee API and Bee Debug API. These versions **does follow**
```
Comment (acud) [inline README.md:30]:
```
Please note that Bee's API versioning (denoted in our OpenAPI [specifications](link)) **does follow** Semantic Versioning.
```

**Review by acud — CHANGES_REQUESTED** [review]
Nice! I have some minor comments

File: README.md (hunk 1 above)
Comment (AuHau) [reply README.md:26]:
```suggestion
There are two versioning schemes used in Bee that you should be aware of. The main Bee version does **NOT** follow
```

File: README.md (hunk 2 above)
Comment (AuHau) [reply README.md:27]:
```suggestion
strict Semantic Versioning. Bee hosts different peer-to-peer wire protocol implementations and individual protocol breaking changes would necessitate a bump in the major part of the version. Breaking changes are expected with bumps of the minor version component. New (backward-compatible) features and bug fixes are expected with a bump of the patch component. Major version bumps are reserved for significant changes in Swarm's incentive structure.

```

File: README.md (hunk 3 above)
Comment (AuHau) [reply README.md:30]:
```suggestion
The second set of versions that are important are the Bee's API versions (denoted in our [Bee](https://github.com/ethersphere/bee/blob/master/openapi/Swarm.yaml) and [Bee Debug](https://github.com/ethersphere/bee/blob/master/openapi/SwarmDebug.yaml) OpenAPI specifications). These versions **do follow** 
```

File: README.md [hunk 4]
```diff
@@ -21,6 +21,16 @@ inability to use, or your interaction with other nodes or the software.
 
 Our documentation is hosted at https://docs.ethswarm.org.
 
+## Versioning
+
+There are two versioning schemes used in Bee that you should be aware of. The main Bee version does **NOT** follow
+strict Semantic Versioning. Bee hosts different peer-to-peer wire protocol implementations and individual protocol breaking changes would necessitate a bump in the major part of the version. Breaking changes are expected with bumps of the minor version component. New (backward-compatible) features and bug fixes are expected with a bump of the patch component. Major version bumps are reserved for significant changes in Swarm's incentive structure.
+
+In this light minor versions can include breaking changes, but patch versions will be kept only for fixes.
```
Comment (AuHau) [inline README.md:29]:
```suggestion
```

**Review by acud — APPROVED** [review]

**Pull request merged**